src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -115,19 +118,29 @@ public static IEnumerable<Method> BuildJsonSerializationMethods(JsonObjectSerial
                 }
             }
 
+            if (iJsonModelInterface == null && iPersistableModelTInterface == null && Configuration.UseModelReaderWriter)
+            {
+                // void JsonModelWriteCore(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+                yield return BuildJsonModelWriteCoreMethod(json, null, hasInherits, isSealed);
+            }
+
             if (iJsonModelInterface is not null && iPersistableModelTInterface is not null)
             {
                 var typeOfT = iJsonModelInterface.Arguments[0];
                 var model = typeOfT.Implementation as SerializableObjectType;
                 Debug.Assert(model != null, $"{typeOfT} should be a SerializableObjectType");
 
+                // void JsonModelWriteCore(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+                var jsonModelWriteCore = BuildJsonModelWriteCoreMethod(json, iPersistableModelTInterface, hasInherits, isSealed);
```

Comment:
consider to put this line before L121, so that it can be reused for both cases.

Reply:
But it would be a waste if neither of these two conditions meets.